utils/preferences/tests/dtf/data_provider.rs

```diff
@@ -0,0 +1,52 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::*;
+use icu_locid::{langid, LanguageIdentifier};
+use icu_preferences::extensions::unicode::keywords;
+use tinystr::tinystr;
+
+struct DefaultPrefs {
+    pub und: DateTimeFormatResolvedPreferences,
+    pub list: &'static [DateTimeFormatResolvedPreferences],
+}
+
+const DEFAULT_PREFS: DefaultPrefs = DefaultPrefs {
```

Comment:
Thought: I think most components won't want to use this exact pattern for DefaultPrefs; they will likely want to load their data first and then populate the ResolvedPreferences based on that. It's fine to do this hard. Consider adding a comment saying that this is just a test, not the way we expect all clients to write DefaultPrefs.